woocommerce/woocommerce-android
# Reorder cards

Part of: #10703

### Why
As part of Expanded Analytics Hub milestone 2, we plan to add support for customizing the cards we display on the Analytic Hub screen.

### Description
This PR adds support for reordering the analytics cards. It does that by adding drag-and-drop support for the compose lazy list and updating the order of cards on the ViewModel. Because we already were persisting the list on the device, not further changes were required.

### Testing instructions
TC1
1. Open the app
2. Tap on See more
3. Tap on the pencil icon to edit the screen configuration
4. Reorder the screen by dragging and dropping cards using the handle icon (=)
5. Check that the list order changes
6. Tap on save to persist the order
7. Tap on the pencil icon again to open the Customize Analytics screen 
8. Check that the order matches the same order as in step 6

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/18119390/1ee78ddc-242c-4bb9-a6c9-509fcdc4cef8
